wire-ios-data-model/Source/ManagedObjectContext/Migration/CoreDataMessagingMigrationVersion.swift

```diff
@@ -130,4 +129,70 @@ enum CoreDataMessagingMigrationVersion: String, CaseIterable {
             subdirectory: Constant.modelDirectory
         )
     }
+
+    var preMigrationAction: CoreDataAction? {
+        switch self {
+        case .version2_111:
+            return RemoveDuplicatePreAction()
+
+        default:
+            return nil
+        }
+    }
+
+    var postMigrationAction: CoreDataAction? {
+        switch self {
+        case .version2_111:
+            return PrefillPrimaryKeyAction()
+
+        default:
+            return nil
+        }
+    }
+}
+
+extension CoreDataMessagingMigrationStep {
+    enum Error: Swift.Error {
+        case wrongModel
+    }
+
+    func runPreMigrationStep(for storeURL: URL) throws {
```

Comment:
I think the `enum CoreDataMessagingMigrationStep` is not the right place to implement a migration, it should not be its responsibility.